### ISSUE TITLE
AppName vs AppTitle?

### ISSUE THREAD

**Issue by SeanAWalsh** [NetSparkleUpdater/NetSparkle#182]
There are places in the UI where the Application Name is used and, in my opinion, the Application Title would be better. So if my product has an application title called "TheGreatProduct.exe" and an ApplicationTitle of "The Great Product", I currently get told

"TheGreatProduct 1.3.49 is now available (you have 1.3.48). Would you like to download it now?" or when downloading, the progress window says "Downloading TheGreatProduct...". I'd prefer: "The Great Product 1.3.49 is now available (you have 1.3.48). Would you like to download it now?" and "Downloading The Great Product...". A minor point but I think it adds to the professionalism of the upgrade experience.

To this end, I made the following changes (line numbers approximate as knocked out sligtly by additions, but if lines are added in sequence should tie up ok)...

---------------------------------------------------------------------
1. Add ApplicationTitle to Netsparkle\Configurations\Configuration object and read it it from AssemblyTitle. For those situations where AssemblyTitle is not set, use AssemblyProduct (so no change in behaviour!)

namespace NetSparkleUpdater.Configurations
{
    public abstract class Configuration
    {
	/// <summary>
        /// The application name
        /// </summary>
        public string ApplicationName { get; protected set; }		(line 21, add 4 below to declare ApplicationTitle)
        /// <summary>
        /// The application title
        /// </summary>
        public string ApplicationTitle { get; protected set; }
		
		...
            try
            {
                // set some value from the binary
                AssemblyAccessor = assemblyAccessor;
		ApplicationName = assemblyAccessor.AssemblyProduct;		(line 81, add line below to assign ApplicationTitle)
		ApplicationTitle = assemblyAccessor.AssemblyTitle.Length > 0 ? assemblyAccessor.AssemblyTitle : assemblyAccessor.AssemblyProduct;
		...
	
---------------------------------------------------------------------
2. Add AppTitle to Netsparkle\AppCastItem object

    [Serializable]
    public class AppCastItem : IComparable<AppCastItem>
    {
			...
        public string AppName { get; set; }								(line 18, add 3 below to declare AppTitle)
        /// <summary>
        /// The application title
        /// </summary>
        public string AppTitle { get; set; }

        /// <summary>
        /// Parse item Xml Node to AppCastItem
        /// </summary>
        /// <param name="installedVersion">Currently installed version</param>
        /// <param name="applicationName">Application name</param>		(line 174, add 1 below to keep documentation happy)
        /// <param name="applicationTitle">Application title</param>
        /// <param name="castUrl">The url of the appcast</param>
        /// <param name="item">The item XML node</param>
        /// <param name="logWriter">logwriter instance</param>
        /// <returns>AppCastItem from Xml Node</returns>				(add applicationTitle as arg to Parse function)
        public static AppCastItem Parse(string installedVersion, string applicationName, string applicationTitle, string castUrl, XElement item, ILogger logWriter)
        {

            var newAppCastItem = new AppCastItem()
            {
                AppVersionInstalled = installedVersion,
                AppName = applicationName,								(line 186, add 1 below to initialise AppTitle)
                AppTitle = applicationTitle,
		...

---------------------------------------------------------------------
3. In NetSparkle\AppCastHandlers\XMLAppCast, pass the ApplicationTitle from the Configuration to the AppCastItem Parse call
line 166: var currentItem = AppCastItem.Parse(_config.InstalledVersion, _config.ApplicationName, _config.ApplicationTitle, _castUrl, item, _logWriter);

And finally use the AppTitle where it's better than the AppName...

---------------------------------------------------------------------
4. NetSparkle.UI.WinForms.NetCore\DownloadProgressWindow
line 38: lblHeader.Text = lblHeader.Text.Replace("APP", item.AppTitle + " " + item.Version);

---------------------------------------------------------------------
5. NetSparkle.UI.WinForms.NetCore\UpdateAvailableWindow
line 70: lblHeader.Text = lblHeader.Text.Replace("APP", item != null ? item.AppTitle : "the application");

---------------------------------------------------------------------
6. NetSparkle.UI.WinForms.NetFramework\DownloadProgressWindow
line 38: lblHeader.Text = lblHeader.Text.Replace("APP", item.AppTitle + " " + item.Version);

---------------------------------------------------------------------
7. NetSparkle.UI.WinForms.NetFramework\UpdateAvailableWindow
line 70: lblHeader.Text = lblHeader.Text.Replace("APP", item != null ? item.AppTitle : "the application");
line 85: lblInfoText.Text = string.Format("{0} {3} is now available (you have {1}). Would you like to {2} it now?", item.AppTitle, versionString, downloadInstallText, item.Version);

Off to learn Git to see if I can turn this into a pull request... :)

**Comment by Deadpikle** [comment]
I think your overall code is fine, and the changes you've made don't break things, which is good; however, I'm not sure that I agree with the change.

According to [this SO post](https://stackoverflow.com/a/23145113/3938401), `AssemblyTitle` is the `File Description`, not the name of the product. This is also reflected when you right-click an EXE and look at the Properties tab:

![WPF](https://user-images.githubusercontent.com/5092399/113526227-3b1fde80-9587-11eb-8916-44ed22db3222.png)

As a normal user, it would make much more sense for the name to show up in NetSparkleUpdater to be the same as the product name that shows up in the Windows properties window.

HOWEVER, in the same SO post [here](https://stackoverflow.com/a/35159325/3938401), the `AssemblyTitle` shows up in Task Manager as the app title!

So.........Windows is rather confusing, here.

In the docs:

* `AssemblyProductAttribute` -> "Defines a **product name** custom attribute"
* `AssemblyTitleAttribute` -> "Specifies a **description** for an assembly."

I'm not sure what to do. Windows' use of things sort of conflicts with the docs.

I'm leaning on keeping things as they are, but I'm open to arguments the other direction. 🤔

**Comment by SeanAWalsh** [comment]
After I posted the issue, I did what you did and looked at the usage of these things and, you're right, it's all over the shop. And every single example fails to address my problem because they all use YourAssemblyTitle and YourApplication (i.e. Camel Case no spaces) thus not addressing the problem. Checking excel.exe I get
![image](https://user-images.githubusercontent.com/12700766/113875268-79e3ad80-97ae-11eb-8a16-cbcfd28f27d6.png)
I agree with you. Leave things as they sit and I'll adjust my Product Name to include spaces...
